### ISSUE TITLE
Result of baseline

### ISSUE THREAD

**Issue by ghost** [erickrf/multiffn-nli#8]
First of all, thanks for your work. I tried to run it on my own data set. After 10 epochs, the validation accuracy is 0.75. I wonder what is the best result you get in your own experiment and how many words you have used? Thanks!

**Comment by erickrf** [comment]
The accuracy completely depends on the data set! With the SNLI, I can get slightly over 85%. 
What exactly fo you mean with how many words? The maximum sentence size? Vocabulary size?

**Comment by ghost** [comment]
@erickrf  Yes, I mean vocabulary size. What is the exact vocab size you have set?

**Comment by ghost** [comment]
BTW, when I run on a binary classification task, there is `nan` loss error. And what I used is your latest version of code. 

    Creating model
    Starting training
    0 completed epochs, 100 batches Average training batch loss: 1.009225   Validation loss: 0.658700       Validation accuracy: 0.576273   (saved model)
    0 completed epochs, 200 batches Average training batch loss: 0.596674   Validation loss: 0.645678       Validation accuracy: 0.635591   (saved model)
    0 completed epochs, 300 batches Average training batch loss: 0.621308   Validation loss: 0.619195       Validation accuracy: 0.665921   (saved model)
    0 completed epochs, 400 batches Average training batch loss: 0.611059   Validation loss: 0.624303       Validation accuracy: 0.665442
    0 completed epochs, 500 batches Average training batch loss: 0.590632   Validation loss: 0.624962       Validation accuracy: 0.643006
    0 completed epochs, 600 batches Average training batch loss: 0.587246   Validation loss: 0.596960       Validation accuracy: 0.681390   (saved model)
    0 completed epochs, 700 batches Average training batch loss: 0.592703   Validation loss: 0.600708       Validation accuracy: 0.685129
    0 completed epochs, 800 batches Average training batch loss: 0.585967   Validation loss: 0.597229       Validation accuracy: 0.690435
    0 completed epochs, 900 batches Average training batch loss: 0.564621   Validation loss: 0.606148       Validation accuracy: 0.674391
    0 completed epochs, 1000 batches        Average training batch loss: 0.578790   Validation loss: 0.586459       Validation accuracy: 0.699128   (saved model)
    0 completed epochs, 1100 batches        Average training batch loss: 0.571691   Validation loss: 0.599395       Validation accuracy: 0.672793
    0 completed epochs, 1200 batches        Average training batch loss: 0.567564   Validation loss: 0.595321       Validation accuracy: 0.679089
    0 completed epochs, 1300 batches        Average training batch loss: 0.568486   Validation loss: 0.584527       Validation accuracy: 0.696827   (saved model)
    0 completed epochs, 1400 batches        Average training batch loss: 0.566347   Validation loss: 0.578138       Validation accuracy: 0.700630   (saved model)
    0 completed epochs, 1500 batches        Average training batch loss: 0.552183   Validation loss: 0.575002       Validation accuracy: 0.703410   (saved model)
    0 completed epochs, 1600 batches        Average training batch loss: 0.546450   Validation loss: 0.576786       Validation accuracy: 0.693535
    0 completed epochs, 1700 batches        Average training batch loss: 0.572475   Validation loss: 0.573494       Validation accuracy: 0.707086   (saved model)
    0 completed epochs, 1800 batches        Average training batch loss: 0.534633   Validation loss: 0.558439       Validation accuracy: 0.712903   (saved model)
    0 completed epochs, 1900 batches        Average training batch loss: 0.550429   Validation loss: 0.556030       Validation accuracy: 0.719550   (saved model)
    0 completed epochs, 2000 batches        Average training batch loss: 0.558161   Validation loss: 0.563393       Validation accuracy: 0.705776
    0 completed epochs, 2100 batches        Average training batch loss: 0.544322   Validation loss: 0.549832       Validation accuracy: 0.724696   (saved model)
    0 completed epochs, 2200 batches        Average training batch loss: nan        Validation loss: nan    Validation accuracy: 0.472338

It is quite strange that the changing of number of classes can lead to the `nan` loss. Do you have any idea? Thanks in advance!

**Comment by erickrf** [comment]
As for the vocabulary, I use the biggest Glove pretrained vectors, with around 2 million words. 
Your nan might be caused by a high learning rate - that's the first thing to check. The model is learning well for some hundred batches, it isn't necessarily linked to the fact that you have a binary problem.

**Comment by ghost** [comment]
I set the learning rate as you suggested, `0.05`. I also tried to reduce it but same thing happened. I tried setting the `num_class` as 3, and everything works well. It is so strange....
